go/exoservice/exoservice.go

```diff
@@ -95,10 +96,14 @@ func (e *ExoService) receiveMessage(message structs.Message) {
 	if e.messageHandlers[message.Name] == nil {
 		return
 	}
+	if len(e.sessionIds) == 0 {
+		e.sessionIds = map[string]string{}
+	}
+	e.sessionIds[message.ID] = message.SessionID
 	e.messageHandlers[message.Name](Request{
 		Payload: message.Payload,
 		Reply:   e.buildSendHelper(message.ID),
-		Send:    e.buildSendHelper(""),
+		Send:    e.buildSendHelper(message.ID),
```

Comment:
Thoughts on passing sessionId as the second argument into the buildSendHelper instead of storing a map of message id to sessionId?